### PR TITLE
[Merged by Bors] - Fix clippy lints for 1.57

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- New clippy lints with rust 1.57 are failing

## Solution

- Fixed clippy lints following suggestions
- I ignored clippy in old renderer because there was many and it will be removed soon

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
All pretty reasonable lints, and the changes are trivially correct. LGTM.

File: crates/crevice/src/glsl.rs
Comment (DJMcNab) [inline crates/crevice/src/glsl.rs:18]: lol

We could make it a sealed trait.

Comment (mockersf) [reply crates/crevice/src/glsl.rs:18]: I did not want to change too much from upstream crevice, refactoring the doc comment to match the lint seemed better for that

Comment (cart) [reply crates/crevice/src/glsl.rs:18]: Yup until we come to a final decision on whether we (1) use upstream crevice (2) use vendored crevice with additional upstream prs merged (3) fork crevice indefinitely (4) build our own thing, I don't want to touch this code at all.

**Comment by cart** [comment]
Lmao our new `trybuild` tests are failing because `rustc` now adds more indentation in their error message annotations:

image

File: crates/crevice/src/glsl.rs
Comment (DJMcNab) [reply crates/crevice/src/glsl.rs:18]: Should we then instead just ignore all of the lints for this crate instead?

Comment (cart) [reply crates/crevice/src/glsl.rs:18]: I'm open to both solutions (and agree that ignoring lints is probably the better forward facing fix), but I won't block on that.

Comment (cart) [reply crates/crevice/src/glsl.rs:18]: The most important thing is to not meaningfully change the code itself. As long as we aren't doing that, I don't care too much.

**Comment by mockersf** [comment]
> Lmao our new `trybuild` tests are failing

Oh I forgot about those...

**Comment by cart** [comment]
Yeah this is the first time they've introduced noise (which was a concern that came up when they were added). I doubt it will come up much, but we should start mentally tracking how disruptive they are.

File: crates/crevice/src/glsl.rs
Comment (mockersf) [reply crates/crevice/src/glsl.rs]: I ignored clippy in crevice

**Comment by mockersf** [comment]
if it becomes painful we could look into having a "bless" tool that would validate all changes and update the files

**Review by NiklasEi — APPROVED** [review]

**Pull request closed without merging**
